packages/application-shell/src/components/application-shell-authenticated/new-main-navigation-flag-wrapper.tsx

```diff
@@ -2,9 +2,27 @@
 import { useFeatureToggle } from '@flopflip/react-broadcast';
 import { MAIN_NAVIGATION } from '../../feature-toggles';
 
-const NewMainNavigationFlagWrapper = (props: { children: Function }) => {
+type TNewMainNavigationFlagWrapperChildrenProps = {
+  isNewNavigationEnabled: boolean;
+  isNewNavigationEnabledEvaluationReady: boolean;
+};
+
+const NewMainNavigationFlagWrapper = (props: {
+  children: ({
+    isNewNavigationEnabled,
+    isNewNavigationEnabledEvaluationReady,
+  }: TNewMainNavigationFlagWrapperChildrenProps) => JSX.Element;
+}) => {
   const isNewNavigationEnabled = useFeatureToggle(MAIN_NAVIGATION);
-  return props.children(isNewNavigationEnabled);
+  // A temporary workaround for a glitch in computing the aggregated adapter status within useAdapterStatus
```

Comment:
**nit**: So it will easier for us to find pending things to do in the future
```suggestion
  // TODO: A temporary workaround for a glitch in computing the aggregated adapter status within useAdapterStatus
```